### PR TITLE
Add measurement campaign table

### PR DESCRIPTION
**Pull request by cortadocodes**
# Summary
Add support for thew new measurement campaigns table in BigQuery. This allows data to be associated with a measurement campaign that can be given a label and description. The code has also been cleaned up and some bugs fixed.

# Contents ([#108](https://github.com/aerosense-ai/data-gateway/pull/108))

**IMPORTANT:** There are 2 breaking changes.

### New features
- Add measurement campaign data (start time, end time, label, and available sensors) to `session` table when uploading a window

### Enhancements
- Replace `label` field with `session_reference` in existing tables

### Fixes
- Raise warning if data gateway configuration not present in metadata

### Reversions
- 💥 **BREAKING CHANGE:**  Remove ability to provide label via CLI

### Operations
- Use latest conventional commit message checker
- Use new Octue GitHub actions in workflows
- Pin `snok/install-poetry` to avoid missing poetry bug
- Ensure poetry is available within tox
- Use new syntax for setting outputs in GitHub workflow

### Dependencies
- 💥 **BREAKING CHANGE:** Drop `python3.7` support
- Update dependencies to patch some security vulnerabilities
- Use `octue==0.41.0`
- Add `pandas` dev dependency
- Add missing `pyarrow` dev dependency
- Combine old `dev-dependencies` group with new `dev` group
- Regenerate requirements files for Raspberry Pi

### Refactoring
- Remove repetition of method call in `BatchingUploader`
- Factor stopping the gateway out into a method

### Testing
- Avoid timezone issues with timestamp value of 0
- Update `MockBigQueryClient`
- Rename `MockBigQueryClient.rows` attribute to `inserted_rows`
- Use `Mock` instead of `types.SimpleNamespace`
- Update deployment test to check for session creation

---
# Upgrade instructions
<details>
<summary>💥 <b>Remove ability to provide label via CLI</b></summary>

Provide a "label" key mapped to the label  in the "measurement_campaign" dictionary of the configuration file instead.
</details>

<details>
<summary>💥 <b>Drop python3.7 support</b></summary>

Update to `python>=3.8`.
</details>

### CODE REVIEW

File: requirements-pi.txt [hunk 1]
```diff
@@ -1,52 +1,55 @@
...
-six==1.16.0; python_full_version >= "3.7.1" and python_version >= "3.6" and python_full_version < "4.0.0" and (python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version >= "3.6")
-text-unidecode==1.3; python_version >= "3.6"
-twined==0.5.0; python_full_version >= "3.7.1" and python_full_version < "4.0.0" and python_version >= "3.6"
-typing-extensions==4.2.0; python_version < "3.8" and python_version >= "3.7" and python_full_version >= "3.7.1" and python_full_version < "4.0.0"
-urllib3==1.26.9; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.6.0" and python_version < "4" and python_version >= "3.6"
-werkzeug==2.1.2; python_full_version >= "3.7.1" and python_full_version < "4.0.0" and python_version >= "3.7"
-zipp==3.8.0; python_version < "3.8" and python_version >= "3.7" and python_full_version >= "3.7.1" and python_full_version < "4.0.0"
+attrs==22.1.0 ; python_full_version >= "3.7.1" and python_version < "3.10"
```
Comment (cortadocodes) [inline requirements-pi.txt:1]: Update this and the dev equivalent if @thclark confirms it's ok to drop `python3.7` support

**Comment by time-trader** [comment]
We should refactor "session" to measurement_campaign" or something like that, to avoid confusion.

**Comment by cortadocodes** [comment]
Is the sessions table part of this PR relevant anymore as we'll be calculating sessions dynamically using the `RawSignal` class?

File: requirements-pi.txt (hunk 1 above)
Comment (cortadocodes) [reply requirements-pi.txt:1]: @thclark please could you comment on this?

**Review by time-trader — COMMENTED** [review]
Changes to conversion constants and number of sensors..

File: tests/valid_configuration.json [hunk 2]
```diff
@@ -213,7 +216,8 @@
         "Gyro": [1, 1, 1],
         "Mag": [1, 1, 1],
         "Analog Vbat": [1],
-        "Constat": [1, 1, 1, 1]
+        "Constat": [1, 1, 1, 1],
+        "battery_info": [1]
```
[the remark this answers is not in the document]
Comment (cortadocodes) [reply tests/valid_configuration.json:220]: Thanks, I'll open a separate issue to update the rest of the sensors

**Pull request merged**